service/debugger/debugger.go

```diff
@@ -420,6 +428,12 @@ func (d *Debugger) Restart(rerecord bool, pos string, resetArgs bool, newArgs []
 		p, err = d.recordingRun(run)
 		d.recordingDone()
 	} else {
+		if rebuild {
+			err := gobuild.GoBuild(d.processArgs[0], d.config.Packages, d.config.BuildFlags)
```

Comment:
Some error message should be returned if the debugger was started with `attach` or `exec`, also this doesn't work if the debugger was started with `test`.